api/v1alpha1/hostedcluster_types.go

```diff
@@ -267,6 +271,53 @@ type ClusterAutoscaling struct {
 	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// EtcdManagementType is a enum specifying the strategy for managing the clusters etcd instance
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type EtcdManagementType string
+
+const (
+	Managed   EtcdManagementType = "Managed"
+	Unmanaged EtcdManagementType = "Unmanaged"
+)
+
+type EtcdSpec struct {
+	// ManagementType defines how the etcd cluster is managed. Unmanaged means
+	// the etcd cluster is managed by a system outside the hypershift controllers.
+	// Managed means the hypershift controllers manage the provisioning of the etcd cluster
+	// and the operations around it
+	ManagementType EtcdManagementType `json:"managementType"`
+
+	// Managed provides metadata that defines how the hypershift controllers manage the etcd cluster
+	Managed *ManagedEtcdSpec `json:"managed,omitempty"`
```

Comment:
Can we mark Managed/Unmanaged fields as optionals?

Reply:
Both these fields need `// +optional`

Reply:
done